extensions/wikia/PolldaddyTag/PolldaddyTagController.class.php

```diff
@@ -0,0 +1,50 @@
+<?php
+class PolldaddyTagController extends WikiaController {
+
+	const PARSER_TAG_NAME = 'polldaddy';
+	const NAME = 'Polldaddy';
+
+	const DATA_WIKI_WIDGET_ATTRIBUTE = [
+		'data-wikia-widget' => self::PARSER_TAG_NAME,
+	];
+
+	private $helper;
+	private $validator;
+
+	public function __construct() {
+		parent::__construct();
+
+		$this->helper = new WikiaIFrameTagBuilderHelper();
+		$this->validator = new PolldaddyTagValidator();
+	}
+
+	public static function onParserFirstCallInit( Parser $parser ) {
+		$parser->setHook( self::PARSER_TAG_NAME, [ new static(), 'renderTag' ] );
+		return true;
+	}
+
+	public function renderTag( $input, array $args, Parser $parser, PPFrame $frame ) {
+		$isValid = $this->validator->validateAttributes( $args );
+
+		if ( !$isValid ) {
+			return '<strong class="error">' . wfMessage( 'polldaddy-tag-could-not-render' )->parse() . '</strong>';
+		}
+
+		return $this->helper->isMobileSkin() ?
+			Html::rawElement(
+				'a',
+				array_merge( self::DATA_WIKI_WIDGET_ATTRIBUTE, [ 'data-id' => $args['id'] ] )
+			) :
+			Html::rawElement(
+				'span',
+				self::DATA_WIKI_WIDGET_ATTRIBUTE,
```

Comment:
That's cool, but do we have a ticket for refactoring/polishing/unifying all of the widgets? I would add this comment there.